src/main/kotlin/org/jetbrains/research/testspark/helpers/psiHelpers/JavaPsiHelper.kt

```diff
@@ -173,145 +136,313 @@ class JavaPsiHelper : PsiHelperInterface {
+            e.dataContext.getData(CommonDataKeys.CARET)?.caretModel?.primaryCaret ?: return result.toArray()
+
+        val javaPsiClassWrapped = getSurroundingClass(caret.offset) as JavaPsiClassWrapper?
+        val javaPsiMethodWrapped = getSurroundingMethod(caret.offset) as JavaPsiMethodWrapper?
+        val line: Int? = getSurroundingLine(caret.offset)?.plus(1)
+
+        javaPsiClassWrapped?.let { result.add(getClassDisplayName(it)) }
+        javaPsiMethodWrapped?.let { result.add(getMethodDisplayName(it)) }
+        line?.let { result.add(getLineDisplayName(it)) }
+
+        if (javaPsiClassWrapped != null && javaPsiMethodWrapped != null) {
+            log.info(
+                "The test can be generated for: \n " +
+                    " 1) Class ${javaPsiClassWrapped.qualifiedName} \n" +
+                    " 2) Method ${javaPsiMethodWrapped.methodDescriptor}" +
+                    " 3) Line $line",
+            )
+        }
+
+        return result.toArray()
+    }
+
+    override fun getLineDisplayName(line: Int): String {
+        return "<html><b><font color='orange'>line</font> $line</b></html>"
+    }
+
+    override fun getClassDisplayName(psiClass: PsiClassWrapper): String {
+        return if ((psiClass as JavaPsiClassWrapper).isInterface) {
             "<html><b><font color='orange'>interface</font> ${psiClass.qualifiedName}</b></html>"
-        } else if (isAbstractClass(psiClass)) {
+        } else if (psiClass.isAbstractClass) {
             "<html><b><font color='orange'>abstract class</font> ${psiClass.qualifiedName}</b></html>"
         } else {
             "<html><b><font color='orange'>class</font> ${psiClass.qualifiedName}</b></html>"
         }
     }
 
-    /**
-     * Gets the display name of a method, depending on if it is a (default) constructor or a normal method.
-     * This is used when displaying the name of a method in GenerateTestsActionMethod menu entry.
-     *
-     * @param psiMethod the PSI method of interest
-     * @return the display name of the PSI method
-     */
-    private fun getMethodDisplayName(psiMethod: PsiMethod): String {
-        return if (isDefaultConstructor(psiMethod)) {
+    override fun getMethodDisplayName(psiMethod: PsiMethodWrapper): String {
+        return if ((psiMethod as JavaPsiMethodWrapper).isDefaultConstructor) {
             "<html><b><font color='orange'>default constructor</font></b></html>"
         } else if (psiMethod.isConstructor) {
             "<html><b><font color='orange'>constructor</font></b></html>"
-        } else if (isMethodDefault(psiMethod)) {
+        } else if (psiMethod.isMethodDefault) {
             "<html><b><font color='orange'>default method</font> ${psiMethod.name}</b></html>"
         } else {
             "<html><b><font color='orange'>method</font> ${psiMethod.name}</b></html>"
         }
     }
+
+    private fun validateLine(selectedLine: Int, psiMethod: JavaPsiMethodWrapper, psiFile: PsiFile): Boolean {
```

Comment:
This method definitely needs to be commented on and likely renamed, as its purpose is unclear, and its implementation is not that simple.

Reply:
Actually, I think that this function is redundant, because we already take a method that surrounds the line.
@arksap2002, this is your part of the code, that do you think?

Reply:
this fuction is already deleted